### ISSUE TITLE
automatic differentiation

### ISSUE THREAD

**Issue by ih4cku** [ih4cku/blog#92]
- _Deep Learning Book_ Chapter 6.5.9
- [Automatic Differentiation, A.K.A please save me from backprop!](https://domluna.me/automatic-differentiation-a.k.a-please-save-me-from-backprop/)
- [How auto-differentiation of python autograd (or torch autograd) compares to symbolic differentiation of theano and tensorFlow?](https://www.quora.com/How-auto-differentiation-of-python-autograd-or-torch-autograd-compares-to-symbolic-differentiation-of-theano-and-tensorFlow)
- [Does tensorflow use automatic or symbolic gradients?](http://stackoverflow.com/questions/36370129/does-tensorflow-use-automatic-or-symbolic-gradients)
- [Automatic Differentiation in Deep/Machine Learning Platforms](https://www.reddit.com/r/MachineLearning/comments/36zfgd/automatic_differentiation_in_deepmachine_learning/)
- [How auto-differentiation of python autograd (or torch autograd) compares to symbolic differentiation of theano and tensorFlow?](https://www.quora.com/How-auto-differentiation-of-python-autograd-or-torch-autograd-compares-to-symbolic-differentiation-of-theano-and-tensorFlow)
- [Autograd for Torch](https://blog.twitter.com/2015/autograd-for-torch)
